internal/vds/vds.cpp

```diff
@@ -416,32 +380,28 @@ void fetch_horizon(
 
             auto const cdp = surface.coordinate(row, col);
 
-            auto ij = transform.WorldToIJKPosition({cdp.x, cdp.y, 0});
-            auto k  = transform.AnnotationToIJKPosition({0, 0, depth});
-
-            /* OpenVDS' transformers and OpenVDS data request functions have
-             * different definition of where a datapoint is. E.g. A transformer
-             * (To voxel or ijK) will return (0,0,0) for the first sample in
-             * the cube. The request functions on the other hand assumes the
-             * data is located in the center of a voxel. I.e. that the first
-             * sample is at (0.5, 0.5, 0.5). This is a *VERY* sharp edge in the
-             * OpenVDS API and borders on a bug. It means we cannot directly
-             * use the output from the transformers as input to the request
-             * functions.
-             */
-            ij[0] += 0.5;
-            ij[1] += 0.5;
-             k[2] += 0.5;
-
-            if (not inrange(iline, ij[0]) or not inrange(xline, ij[1])) {
+            auto ij = transform.WorldToOpenvdsPosition(
+                {cdp.x, cdp.y, OpenvdsVoxelVector::NOVALUE}
+            );
+            auto k  = transform.AnnotationToOpenvdsPosition(
+                {OpenvdsVoxelVector::NOVALUE, OpenvdsVoxelVector::NOVALUE, depth}
+            );
+
+            auto coordinate = OpenvdsVoxelVector(k[0], ij[1], ij[2]);
```

Comment:
Just curious, have you tested the performance of this?

The old call stack for these 3 was:
```
transform.WorldToIJKPosition
transform.AnnotationToIJKPosition
```

Now it's
```
transform.WorldToOpenvdsPosition
  WorldToIJKPosition
    OpenVdsVoxelVector
      OpenVDS::DoubleVector3 (copy)

transform.AnnotationToOpenvdsPosition
  AnnotationToIJKPosition
    OpenVdsVoxelVector
      OpenVDS::DoubleVector3 (copy)

OpenvdsVoxelVector
  OpenVDS::DoubleVector3 (copy)
```

for every point in the horizon

Reply:
That seams like a totally reasonable size. 

What is that ,  ~15-25% increase in execution time? Then we have to ask ourselves. Do we accept a hit like that in order to remove some ugly, but not magic, numbers from the code?

Reply:
To answer that question I need a handover from you on where to get something similar to actually desired environment and how you yourself use it!

I think below certain threshold we wouldn't care much if something increases in size.
Like if some operation contributes to the final result 1ms, but after a change would contribute 2ms, that is probably OK.

So after some point the absolute value matters more than the relative one. We need to figure out if we would reach that point of not on proper hardware.

If my machine is 4 times slower then final one, the difference for this horizon would be around-ish 0.1 second.
If user pattern doesn't expect user to request 100s horizons per second, that might it theory be acceptable, if it benefits us enough.